data/sql/updates/pending_db_world/molten-war-golem-distract-immunity.sql

```diff
@@ -0,0 +1,2 @@
+-- Set Distract immunity to Molten War Golem
+UPDATE `creature_template` SET `mechanic_immune_mask`= 8 WHERE `entry` = 8908;
```

Comment:
```suggestion
UPDATE `creature_template` SET `mechanic_immune_mask`=`mechanic_immune_mask` | (8|64|1024), `spellschoolmask`=`spellschoolmask`|8 WHERE `entry` = 8908;
```

flags and bits https://www.azerothcore.org/wiki/sql-standards#queries

tool: https://www.azerothcore.org/flag-checker/

Reply:
I haven't tested for any other immunities, though these would make sense. My source is a clip I'm not sure I can post here. What's their source?

Reply:
> I haven't tested for any other immunities, though these would make sense. My source is a clip I'm not sure I can post here. What's their source?

You can link a clip through a youtube link. Although in this case it's no longer needed

Immunities are tested by casting known spells and seeing if the mob is immune. The linked commit has the spells mentioned as a comment.

Reply:
Okay, let me rephrase: I have no way of testing for immunities at the moment.
The issue with your suggestion would be that you're adding a whole spell school as a way to effectively achieve poison immunity done by vmangos, but that would take away too much from nature casters and hasn't been confirmed.

Reply:
Yes, you're right. I wrongfully assumed it was immune to nature and updated the suggestion. It looks like their immunities are handled differently.
see https://github.com/vmangos/core/commit/4f517f4608664b2ddff1a6038103fb60beadd749
```
// Apply Poison & Disease immunities for Elemental and Mechanical type creaturesAdd commentMore actions
    if (GetCreatureInfo()->type == CREATURE_TYPE_ELEMENTAL ||
        GetCreatureInfo()->type == CREATURE_TYPE_MECHANICAL)
    {
        ApplySpellImmune(0, IMMUNITY_DISPEL, DISPEL_DISEASE, true);
        ApplySpellImmune(0, IMMUNITY_DISPEL, DISPEL_POISON, true);
    }
```
The only reference I find to poison immunity is voidreader in TK. I'm certain The Curator should also be immune to poison. I doubt the above can be applied to 3.3, as not every mob of this type should have this immunity. However, at least a flag in the DB could be set to apply this immunity. This looks like a TODO for a new issue :+1:

Reply:
I think the idea is that poison immunity would have a much larger impact on rogues post vanilla and it's therefor situational, so not necessarily wrong without until observed otherwise.